engine/class_modules/monk/sc_monk.cpp

```diff
@@ -3676,30 +3676,57 @@ struct chi_torpedo_t : public monk_spell_t
 // ==========================================================================
 // Crackling Jade Lightning
 // ==========================================================================
-// Power of the Thunder King TODO: Either
-// a) Copy Warlock's Soul Rot implementation for Drain Soul.
-// b)  channel do 0 damage and use a fake tick_action to do a single instance
-// of aoe damage on each tick, and on the tick_action override
-// action_t::amount_type() to return result_amount_type::DMG_OVER_TIME
 
 struct crackling_jade_lightning_t : public monk_spell_t
 {
+  struct crackling_jade_lightning_aoe_t : public monk_spell_t
+  {
+    crackling_jade_lightning_aoe_t( monk_t *p )
+      : monk_spell_t( p, "crackling_jade_lightning_aoe", p->baseline.monk.crackling_jade_lightning )
+    {
+      dual = background = true;
+      ww_mastery        = true;
+
+      parse_effects( p->talent.windwalker.power_of_the_thunder_king, 0B001U );
```

Comment:
Can use this instead, I plan to replace all manual invocations with the wrapper in the near future if you don't want to.

https://github.com/simulationcraft/simc/blob/04b3ef743e37fc10586763b44cb31dafd15148ad/engine/action/parse_effects.hpp#L143-L174